File: RoomTest/app/src/main/java/com/example/roomtest/DAO/InterfaceDaoCoordenada.java
package com.example.roomtest.DAO;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.roomtest.entidades.Coordenada;

import java.util.List;

@Dao
public interface InterfaceDaoCoordenada {

    @Insert
    public void crearCoordenada(Coordenada al);
    @Delete
    public void borrarCoordenada(Coordenada al);
    @Update
    public void modificarCoordenada(Coordenada al);
    @Query("SELECT * FROM coordenada WHERE id LIKE :id")
    public Coordenada verCoordenada(int id);
    @Query("SELECT * FROM Coordenada WHERE nombre LIKE :nombre")
    public Coordenada verACoordenada(String nombre);
    @Query("SELECT * FROM Coordenada")
    public List<Coordenada> verCoordenada();

    @Query("DELETE FROM coordenada")
    public void nukeTable();
}

File: RoomTestKotlin/app/src/main/java/com/example/roomtestkotlin/modelo/entidades/DAO/InterfaceDaoCoordenada.kt
package com.example.roomtestkotlin.modelo.entidades.DAO

import androidx.room.*
import com.example.roomtestkotlin.modelo.entidades.Lugar

@Dao
interface InterfaceDaoCoordenada {
    @Insert
    fun crearLugar(al: Lugar?)

    @Delete
    fun borrarLugar(al: Lugar?)

    @Update
    fun modificarLugar(al: Lugar?)

    @Query("SELECT * FROM Lugar WHERE id LIKE :id")
    fun verLugar(id: Int): Lugar?

    @Query("SELECT * FROM Lugar WHERE nombre LIKE :nombre")
    fun verACoordenada(nombre: String?): Lugar?

    @Query("SELECT * FROM Lugar")
    fun verCoordenada(): List<Lugar?>?

    @Query("DELETE FROM Lugar")
    fun nukeTable()
}